pkg/hubble/dropeventemitter/dropeventemitter.go

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dropeventemitter
+
+import (
+	"context"
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	typedv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	metaslimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	slimscheme "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/scheme"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	"github.com/cilium/cilium/pkg/identity"
+	client "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+type DropEventEmitter struct {
+	history     map[string]time.Time
+	historySize int
+	interval    time.Duration
+	broadcaster record.EventBroadcaster
+	recorder    record.EventRecorder
+}
+
+func NewDropEventEmitter(interval time.Duration, historySize int, k8s client.Clientset) *DropEventEmitter {
+	emitter := DropEventEmitter{
+		history:     map[string]time.Time{},
+		historySize: historySize,
+		interval:    interval,
+		broadcaster: record.NewBroadcaster(),
+	}
+	emitter.broadcaster.StartRecordingToSink(&typedv1.EventSinkImpl{Interface: k8s.CoreV1().Events("")})
+	emitter.recorder = emitter.broadcaster.NewRecorder(slimscheme.Scheme, v1.EventSource{Component: "cilium"})
+
+	return &emitter
+}
+
+func (e *DropEventEmitter) expireHistory(now time.Time) {
```

Comment:
What do you think of using `k8s.io/client-go/tools/record.EventRecorder`? It implements rate limiting directly.

Reply:
Did that, see above. Thanks for the suggestion.